server/server.ts

```diff
@@ -428,7 +429,263 @@ app.get('/api/compare/all', authMiddleware, async (req, res, next) => {
         highestCorr.highestUserId = comparitorUserId;
       }
     });
-    res.json(highestCorr);
+    if (!highestCorr.highestUserId) {
+      res.json(highestCorr);
+      return;
+    }
+    const comparatorId: number = highestCorr.highestUserId;
```

Comment:
This is the meat of your project, the true value that you are providing over other services. So it's important to make this shine. Potential employers will want to look at this and you'll want them to be impressed!

It's pretty tough logic. I couldn't figure out what algorithm you are implementing by reading the code. I urge you to put a lengthy comment at the top of this function that explains what the code is doing and why that will produce a valid comparison.

If necessary, pull this function into another file (using Express Router) so you can isolate this sophisticated logic and document it thoroughly.